### DISCUSSION TITLE
Passing `schema_extra` to Field has changed in Pydantic 2.0 but not in SQLModel.

### DISCUSSION THREAD

**Discussion by HarshTrivedi** [fastapi/sqlmodel#833] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
No code to show. See the description.
```

### Description

PydanticV1's `Field` class supported passing arbitrary keyword arguments to the JSON schema. In SQLModel counterpart, it was supported by passing `schema_extra` (see [this](https://github.com/tiangolo/sqlmodel/blob/main/sqlmodel/main.py#L306)).

PydanticV2 still supports passing extra things to the JSON schema but in a slightly different way. One has to pass a dictionary in the `json_schema_extra` argument (see [this](https://docs.pydantic.dev/latest/migration/#changes-to-pydanticfield)).

Currently, SQLModel does not reflect the corresponding change in its Field implementation. From what I can tell, it's simply about making one line change to the call [here](https://github.com/tiangolo/sqlmodel/blob/main/sqlmodel/main.py#L343) (call to Pydantic's FieldInfo).

```python
# from:
**current_schema_extra,
# to:
json_schema_extra=current_schema_extra,
```

For consistency with Pydantic, it may also be better to rename `schema_extra` to `json_schema_extra`.

### Operating System

macOS

### SQLModel Version

0.0.16

### Python Version

3.11.0

**Comment by HarshTrivedi** [comment]
Okay, it's been over a week and I am fairly positive this is a bug. So tagging @tiangolo, @maru0123-2004 @daniil-berg, @raphaelgibson based on who have made [commits to the lines surrounding the bug](https://github.com/tiangolo/sqlmodel/blame/main/sqlmodel/main.py#L343). I'll appreciate the help, thank you!

**Comment by HarshTrivedi** [comment]
@tiangolo  Gentle reminder here.

**Reply by chenkovsky** [reply]
there's a workaround
```
schema_extra={
        "json_schema_extra": {
            "x-yy-zz": "zz"
        }
    }
```

**Comment by maxsgcf** [comment]
Same issue. There is a prevision of when it would be changed?

**Comment by KimigaiiWuyi** [comment]
Same issue.

**Comment by maru0123-2004** [comment]
Sorry for late replying.
**SQLModel's** `schema_extra` args is not directly passing to **Pydantic(v1)'s** `schema_extra` args.
I think it means **SQLModel's** `schema_extra` is not wants to restrict usage for **Pydantic(v1)'s** `schema_extra`, but wants passing more "extra" info to SQLModel's schema.

There may exists problem for argument naming, but it is not bad than mine.
If there are use case with Pydantic's `schema_extra` or `json_schema_extra`, its may be better adding `json_schema_extra` args.

My contribution is not affected to this problem, I think. But if there are more problem having relation with my code, please mention again.
Thanks.

P.S. Sorry for bad my English.

**Reply by KimigaiiWuyi** [reply]
> Sorry for late replying. **SQLModel's** `schema_extra` args is not directly passing to **Pydantic(v1)'s** `schema_extra` args. I think it means **SQLModel's** `schema_extra` is not wants to restrict usage for **Pydantic(v1)'s** `schema_extra`, but wants passing more "extra" info to SQLModel's schema.
> 
> There may exists problem for argument naming, but it is not bad than mine. If there are use case with Pydantic's `schema_extra` or `json_schema_extra`, its may be better adding `json_schema_extra` args.
> 
> My contribution is not affected to this problem, I think. But if there are more problem having relation with my code, please mention again. Thanks.
> 
> P.S. Sorry for bad my English.

I created this [Pull Request](https://github.com/tiangolo/sqlmodel/pull/1035) to solve this problem, it doesn't affect the original configuration, is backward compatible while at the same time and keeps the parameters synchronized with Pydantic V2, which I think should solve part of the parameter naming confusion.

**Comment by sneakers-the-rat** [comment]
The behavior is even more cursed if you try and work around it by using `pydantic.Field` rather than `sqlmodel.Field`

```python
from sqlmodel import SQLModel
from pydantic import Field

class MyModel(SQLModel):
    x: int = Field(json_schema_extra={'key': 'value'})

MyModel.model_fields['x']
# FieldInfo(..., json_schema_extra={'schema_extra': {'json_schema_extra': {'key': 'so'}}})
```

**Comment by YuriiMotov** [comment, marked as answer]
AFAIK it's not documented, but `schema_extra` parameter of SQLModel's `Field` is for passing parameters that are not present in the `Field`, but supported by Pydantic's `Field`.

For example, there is no `validation_alias` parameter, but you can pass it the following way:
```py
id_: in = Field(schema_extra={"validation_alias": "item_id"})
```
